friends/src/App.js

```diff
@@ -0,0 +1,99 @@
+import React, { Component } from 'react';
+import { Route } from 'react-router-dom';
+import axios from 'axios';
+import FriendList from './components/FriendList';
+import FriendForm from './components/FriendForm';
+
+class App extends Component {
+  state = {
+    friends: [],
+    id: '',
+    name: '',
+    age: '',
+    email: ''
+  };
+
+  componentDidMount() {
+    this.fetchFriends();
+  }
+
+  fetchFriends() {
+    axios
+      .get('http://localhost:5000/friends')
+      .then(({ data }) => this.setState({ friends: data }))
+      .catch(err => console.error(err));
+  }
+
+  handleFriendSubmit = event => {
+    event.preventDefault();
+
+    const { name, age, email, id } = this.state;
+    const url = id
+      ? `http://localhost:5000/friends/${id}`
+      : 'http://localhost:5000/friends';
+
+    axios[id ? 'put' : 'post'](url, { name, age, email })
+      .then(({ data }) =>
+        this.setState({
+          friends: data,
+          id: '',
+          name: '',
+          age: '',
+          email: ''
```

Comment:
something interesting you might like to try to avoid repetition like this when resetting the input field state is to hold all 4 of the relevant values outside the component in a separate object -- `initialFriend ={id, etc..}` which you can then set on state in your initial componentDidMount (if stored as an object such as `friend` or `friendData` or something along those lines). Then when you need to reset the field values here, in your setState you can reset it with just `friendData : initialFriend` and save yourself some typing, especially if there are multiple methods you invoke that you want to reset them